### DISCUSSION TITLE
On LINCS warnings and the numerical stability of Martini 3 models

### DISCUSSION THREAD

**Discussion by yuqingxiong96** [ricalessandri/Martini3-small-molecules#47] [Q&A] [answered]
Hello, when I finished my martini3 model of small molecules, I performed a molecular dynamics simulation with a time step of 10fs. When the simulation is at step 31144925, the simulation terminates due to a large number of LINCS WARNINGS and an error Bond length is not finite. I checked the trajectory through vmd and found that the conformation of small molecules in the last frame 31145 was normal, but in 31144 frame, the two particles bound by constrain were at atopposite sides of the box due to periodicity. I don't know if that's what caused the system to explode. Can you give me some help to find the cause of the problem?

**Comment by zhangtrzhangtr** [comment, marked as answer]
I have encountered this issue before, and the reason for the infinitely increasing bond lengths might be multifaceted. Sometimes, specific versions of Gromacs may cause this issue due to the way they handle LINCS. Additionally, excessively high force constants on certain bonds can also lead to this problem. Furthermore, inappropriate exclusions settings in the top file may also be a cause. These are just some of my experiences, haha

**Reply by ricalessandri** [reply]
Yes, @zhangtrzhangtr . Let me highlight that specific Gromacs versions have been indeed somewhat problematic with constraints and virtual sites, with some versions being less numerically stable than others (e.g., see [here](https://github.com/ricalessandri/Martini3-small-molecules/discussions/45#discussioncomment-8877470)). Apparently, Gromacs version 2023.4 seems to fix all the issues, so I'd recommend using that if possible.

The numerical stability of models is also discussed in [the M3 small molecule paper](https://doi.org/10.1002/adts.202100391), section "2.1.3. Advanced Model Design Strategies". Although not comprehensive, it may give you some more ideas and starting points @yuqingxiong96.
